### DISCUSSION TITLE
Logging cFS messages to a file in BootesRC2 -- Is DataStore compatible?

### DISCUSSION THREAD

**Discussion by bdcmu** [nasa/cFS#544] [Q&A] [answered]
I am currently working with the BootesRC2 release of cFS and I need a way to log Software Bus messages to the filesystem. My first thought was to use DataStore, but I did not see any compatibility information about the older versions in the release notes. Is DataStore 2.5.2 or 2.5.1 compatible with BootesRC2?

I was able to get DataStore compiling by adding in the missing cfs_lib files but I seem to be getting a strange error on DataStore startup:
```
1980-012-14:03:20.29428 ES Startup: Loading file: /cf/ds.so, APP: DS_APP
1980-012-14:03:20.29445 ES Startup: DS_APP loaded and created
EVS Port1 0/1/CFE_SB 4: CreatePipeErr:OS_QueueCreate returned -11,app DS_APP
EVS Port1 0/1/DS_APP 2: Unable to create input pipe, err = 0xCA000005
EVS Port1 0/1/DS_APP 3: Application terminating, err = 0xCA000005
1980-012-14:03:20.29461 DS application terminating, err = 0xCA000005
1980-012-14:03:20.29463 Application DS_APP called CFE_ES_ExitApp
```
All of my other cFS apps are working and I have not changed the failing line of code from what is in the repo: (looks okay to me)
`Result = CFE_SB_CreatePipe(&DS_AppData.InputPipe,  DS_APP_PIPE_DEPTH, DS_APP_PIPE_NAME);`

If DataStore is incompatible with BootesRC2, are there any alternative logging apps that exist?

**Comment by skliper** [comment, marked as answer]
`OS_QueueCreate` error code -11 is "invalid size" (at least in the main branch):
https://github.com/nasa/osal/blob/6e6afb4d3ea611f5d11e4e94a5adee595cc08b99/src/os/inc/osapi-error.h#L107

Maybe DS_APP_PIPE_DEPTH is > whatever your OS_QUEUE_MAX_DEPTH is set to?

**Reply by bdcmu** [reply]
Yes, that was it...thanks for the help!
